frontend/src/views/MockExams/MockExamsQuestion.tsx

```diff
@@ -0,0 +1,309 @@
+}
+
+interface MediaInterface {
+  mediaURL: string | null;
+  hasVideo: boolean | undefined;
+  onVideoEnd: () => void;
+}
+
+export const MockExamsQuestion = ({
+  question,
+  onQuestionSubmit,
+  isLastQuestion,
+}: MockExamQuestionInterface) => {
+  const { answers } = question;
+  const selectedAnswerRef = useRef<number | undefined>(undefined);
+  const [selectedAnswer, setSelectedAnswerState] = useState<number>();
+  const [progress, setProgress] = useState(0);
+  const [state, startReadingQuestion, startAnsweringQuestion, startVideo] =
+    useQuestionState();
+  const [elementaryQuestionsAmount, setElementaryQuestionsAmount] = useState(0);
+  const [advancedQuestionsAmount, setAdvancedQuestionsAmount] = useState(0);
+
+  const setSelectedAnswer = (value: number | undefined) => {
+    setSelectedAnswerState(value);
+    selectedAnswerRef.current = value;
+  };
+  const maxTimeInMs = useMemo(() => {
+    const maxTime =
+      state.type === 'reading'
+        ? question.type.timeToReadTheQuestion
+        : question.type.timeToAnswer;
+    return maxTime * 1000;
+  }, [
+    question.type.timeToAnswer,
+    question.type.timeToReadTheQuestion,
+    state.type,
+  ]);
+
+  const hasMedia = useMemo(() => question.mediaReference !== null, [question]);
+  const hasVideo = useMemo(() => {
+    return hasMedia && question.mediaReference?.endsWith('.wmv');
+  }, [question.mediaReference, hasMedia]);
+
+  const submitQuestion = useCallback(() => {
+    onQuestionSubmit(question.id, selectedAnswerRef.current);
+  }, [onQuestionSubmit, question.id]);
+
+  const nextState = useCallback(() => {
+    if (state.type === 'reading') {
+      if (hasVideo) {
+        startVideo();
+      } else {
+        startAnsweringQuestion();
+      }
+    } else {
+      submitQuestion();
+    }
+  }, [
+    hasVideo,
+    startAnsweringQuestion,
+    startVideo,
+    state.type,
+    submitQuestion,
+  ]);
+
+  useEffect(() => {
+    setSelectedAnswer(undefined);
+    if (question.type.scope === 'Podstawowy') {
+      startReadingQuestion();
+      setElementaryQuestionsAmount((prev) => prev + 1);
+    } else {
+      startAnsweringQuestion();
+      setAdvancedQuestionsAmount((prev) => prev + 1);
+    }
+  }, [question, startAnsweringQuestion, startReadingQuestion]);
+
+  useEffect(() => {
+    let currentTimePassed = 0;
+    const step = maxTimeInMs / 500;
+    let timer: NodeJS.Timeout;
+    if (state.type === 'video') {
+      setProgress(0);
+    } else {
+      timer = setInterval(() => {
+        currentTimePassed += step;
+        const newProgress = (currentTimePassed / maxTimeInMs) * 100;
+        if (newProgress >= 100) {
+          nextState();
+        } else {
+          setProgress((currentTimePassed / maxTimeInMs) * 100);
+        }
+      }, step);
+    }
+
+    return () => {
+      clearInterval(timer);
+    };
+  }, [
+    nextState,
+    maxTimeInMs,
+    state,
+    question.type.timeToAnswer,
+    question.type.timeToReadTheQuestion,
+  ]);
```

Comment:
It'd be better to use browser types, rather the node's (I know it's a bit fucked up that it works like this by default 😕)

```suggestion
  useEffect(() => {
    let currentTimePassed = 0;
    const step = maxTimeInMs / 500;
    let timer: number;
    if (state.type === 'video') {
      setProgress(0);
    } else {
      timer = window.setInterval(() => {
        currentTimePassed += step;
        const newProgress = (currentTimePassed / maxTimeInMs) * 100;
        if (newProgress >= 100) {
          nextState();
        } else {
          setProgress((currentTimePassed / maxTimeInMs) * 100);
        }
      }, step);
    }

    return () => {
      window.clearInterval(timer);
    };
  }, [
    nextState,
    maxTimeInMs,
    state,
    question.type.timeToAnswer,
    question.type.timeToReadTheQuestion,
  ]);
```